pkg/sql/schemachanger/schemachanger_test.go

```diff
@@ -490,6 +490,66 @@ func TestSchemaChangeWaitsForConcurrentSchemaChanges(t *testing.T) {
 			"defaultdb", "t", expectedKeyCount)
 	}
 
+	typeSchemaChangeFunc := func(t *testing.T, modeFor1stStmt, modeFor2ndStmt sessiondatapb.NewSchemaChangerMode) {
+		ctx, cancel := context.WithCancel(context.Background())
+		dropTypeChan := make(chan struct{})
+		addTypeValueChan := make(chan struct{})
+		var closeAlterPKChanOnce sync.Once
+		schemaChangeWaitDetected := false
+
+		var params base.TestServerArgs
+		params.Knobs = base.TestingKnobs{
+			SQLDeclarativeSchemaChanger: &scexec.TestingKnobs{
+				// If the blocked schema changer is from legacy schema changer, we let
+				// it hijack this knob (which is originally design for declarative
+				// schema changer) if `stmt` is nil.
+				WhileWaitingForConcurrentSchemaChanges: func(stmts []string) {
+					if (len(stmts) == 1 && strings.Contains(stmts[0], "DROP TYPE")) ||
+						stmts == nil {
+						closeAlterPKChanOnce.Do(func() {
+							close(addTypeValueChan)
```

Comment:
nit: These channel names feel backwards? I would expect this one to be the `dropType` channel. Alternatively you could give them past tense names like `typevalueAddedChan`?